Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Ballistic.cs

```diff
@@ -49,10 +49,48 @@ private void OnBallisticInteractUsing(EntityUid uid, BallisticAmmoProviderCompon
             return;
 
         if (GetBallisticShots(component) >= component.Capacity)
+        {
+            Popup(Loc.GetString("gun-ballistic-transfer-target-full", ("entity", args.Target)), uid, args.User);
             return;
+        }
+
+        if (EntityManager.HasComponent<SpeedLoaderComponent>(args.Used))
+        {
```

Comment:
Why is this doubling the existing speedloaders? It's not even using a whitelist or anything.

Reply:
What do you mean by this? Doubling the existing speedloaders? It is also using a whitelist for anything inserted.
![image](https://github.com/user-attachments/assets/3cbb9327-4e25-46c7-8639-287083711d6c)